# Absenteeism Analysis: Project Overview
    1. Focus on the business problem
    2. Look into the dataset
    3. Apply logistic regression
    4. Create a dataviz
    5. Try to predict absenteeism from work to know whether or not an employee can be expected to be missing for a specific number of hours in a given workday
    6. Have above information in advance can improve our decision-making by reorganizing the work process in a way that will allow us to avoid a lack of productivity and increase the quality of work 
    7. To explore whether a person presenting certain characteristics is expected to be away from work at some point in time or not
    
# Absenteeism 
    Absence from work during normal working hours, resulting in temporary incapacity to execute regular working activity.
    
# This analysis will address ABSENTEEISM at a company during work time
    Problem: 
    1. the business environment of today is more competitive than is used to be, and that leads to increased pressure in the workplace
    2. unachievable business goals
    3. elevated risk of becoming unemployed
    4. becomes detrimental to a person's health

In [2]:
import pandas as pd

In [3]:
# To import the Absenteeism Data in Python
raw_csv_data = pd.read_csv('Absenteeism_data.csv')

In [4]:
raw_csv_data

ID  Reason for Absence        Date  Transportation Expense  \
0    11                  26  07/07/2015                     289   
1    36                   0  14/07/2015                     118   
2     3                  23  15/07/2015                     179   
3     7                   7  16/07/2015                     279   
4    11                  23  23/07/2015                     289   
..   ..                 ...         ...                     ...   
695  17                  10  23/05/2018                     179   
696  28                   6  23/05/2018                     225   
697  18                  10  24/05/2018                     330   
698  25                  23  24/05/2018                     235   
699  15                  28  31/05/2018                     291   

     Distance to Work  Age  Daily Work Load Average  Body Mass Index  \
0                  36   33                  239.554               30   
1                  13   50                  239.554               31   
2                  51   38                  239.554               31   
3                   5   39                  239.554               24   
4                  36   33                  239.554               30   
..                ...  ...                      ...              ...   
695                22   40                  237.656               22   
696                26   28                  237.656               24   
697                16   28                  237.656               25   
698                16   32                  237.656               25   
699                31   40                  237.656               25   

     Education  Children  Pets  Absenteeism Time in Hours  
0            1         2     1                          4  
1            1         1     0                          0  
2            1         0     0                          2  
3            1         2     0                          4  
4            1         2     1                          2  
..         ...       ...   ...                        ...  
695          2         2     0                          8  
696          1         1     2                          3  
697          2         0     0                          8  
698          3         0     0                          2  
699          1         1     1                          2  

[700 rows x 12 columns]

In [5]:
# To make a copy of original data set
df = raw_csv_data.copy() 

In [6]:
df

ID  Reason for Absence        Date  Transportation Expense  \
0    11                  26  07/07/2015                     289   
1    36                   0  14/07/2015                     118   
2     3                  23  15/07/2015                     179   
3     7                   7  16/07/2015                     279   
4    11                  23  23/07/2015                     289   
..   ..                 ...         ...                     ...   
695  17                  10  23/05/2018                     179   
696  28                   6  23/05/2018                     225   
697  18                  10  24/05/2018                     330   
698  25                  23  24/05/2018                     235   
699  15                  28  31/05/2018                     291   

     Distance to Work  Age  Daily Work Load Average  Body Mass Index  \
0                  36   33                  239.554               30   
1                  13   50                  239.554               31   
2                  51   38                  239.554               31   
3                   5   39                  239.554               24   
4                  36   33                  239.554               30   
..                ...  ...                      ...              ...   
695                22   40                  237.656               22   
696                26   28                  237.656               24   
697                16   28                  237.656               25   
698                16   32                  237.656               25   
699                31   40                  237.656               25   

     Education  Children  Pets  Absenteeism Time in Hours  
0            1         2     1                          4  
1            1         1     0                          0  
2            1         0     0                          2  
3            1         2     0                          4  
4            1         2     1                          2  
..         ...       ...   ...                        ...  
695          2         2     0                          8  
696          1         1     2                          3  
697          2         0     0                          8  
698          3         0     0                          2  
699          1         1     1                          2  

[700 rows x 12 columns]

In [7]:
# Print the summary of the Data Frame and check there any missing values
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 700 entries, 0 to 699
Data columns (total 12 columns):
 #   Column                     Non-Null Count  Dtype  
---  ------                     --------------  -----  
 0   ID                         700 non-null    int64  
 1   Reason for Absence         700 non-null    int64  
 2   Date                       700 non-null    object 
 3   Transportation Expense     700 non-null    int64  
 4   Distance to Work           700 non-null    int64  
 5   Age                        700 non-null    int64  
 6   Daily Work Load Average    700 non-null    float64
 7   Body Mass Index            700 non-null    int64  
 8   Education                  700 non-null    int64  
 9   Children                   700 non-null    int64  
 10  Pets                       700 non-null    int64  
 11  Absenteeism Time in Hours  700 non-null    int64  
dtypes: float64(1), int64(10), object(1)
memory usage: 65.8+ KB


# Drop 'ID'

In [8]:
# Drop ID variable from regression equation
df = df.drop(['ID'], axis = 1)

In [9]:
df

Reason for Absence        Date  Transportation Expense  Distance to Work  \
0                    26  07/07/2015                     289                36   
1                     0  14/07/2015                     118                13   
2                    23  15/07/2015                     179                51   
3                     7  16/07/2015                     279                 5   
4                    23  23/07/2015                     289                36   
..                  ...         ...                     ...               ...   
695                  10  23/05/2018                     179                22   
696                   6  23/05/2018                     225                26   
697                  10  24/05/2018                     330                16   
698                  23  24/05/2018                     235                16   
699                  28  31/05/2018                     291                31   

     Age  Daily Work Load Average  Body Mass Index  Education  Children  Pets  \
0     33                  239.554               30          1         2     1   
1     50                  239.554               31          1         1     0   
2     38                  239.554               31          1         0     0   
3     39                  239.554               24          1         2     0   
4     33                  239.554               30          1         2     1   
..   ...                      ...              ...        ...       ...   ...   
695   40                  237.656               22          2         2     0   
696   28                  237.656               24          1         1     2   
697   28                  237.656               25          2         0     0   
698   32                  237.656               25          3         0     0   
699   40                  237.656               25          1         1     1   

     Absenteeism Time in Hours  
0                            4  
1                            0  
2                            2  
3                            4  
4                            2  
..                         ...  
695                          8  
696                          3  
697                          8  
698                          2  
699                          2  

[700 rows x 11 columns]

# Reason for Absence

In [10]:
# To check the minimum number in the 'Reason for Absence' column
df['Reason for Absence'].min()

0

In [11]:
# To check the maximum number in the 'Reason for Absence' column
df['Reason for Absence'].max()

28

In [153]:
# To extract a list of distinct values by ascending order 
sorted(df['Reason for Absence'].unique())

[0,
 1,
 2,
 3,
 4,
 5,
 6,
 7,
 8,
 9,
 10,
 11,
 12,
 13,
 14,
 15,
 16,
 17,
 18,
 19,
 21,
 22,
 23,
 24,
 25,
 26,
 27,
 28]

#### Missing number 20 from the Reason for Absence column

In [12]:
# To convert the nominal values in 'Reason for Absence' column to dummy variables
reason_columns = pd.get_dummies(df['Reason for Absence'])

In [13]:
reason_columns

0   1   2   3   4   5   6   7   8   9   ...  18  19  21  22  23  24  25  \
0     0   0   0   0   0   0   0   0   0   0  ...   0   0   0   0   0   0   0   
1     1   0   0   0   0   0   0   0   0   0  ...   0   0   0   0   0   0   0   
2     0   0   0   0   0   0   0   0   0   0  ...   0   0   0   0   1   0   0   
3     0   0   0   0   0   0   0   1   0   0  ...   0   0   0   0   0   0   0   
4     0   0   0   0   0   0   0   0   0   0  ...   0   0   0   0   1   0   0   
..   ..  ..  ..  ..  ..  ..  ..  ..  ..  ..  ...  ..  ..  ..  ..  ..  ..  ..   
695   0   0   0   0   0   0   0   0   0   0  ...   0   0   0   0   0   0   0   
696   0   0   0   0   0   0   1   0   0   0  ...   0   0   0   0   0   0   0   
697   0   0   0   0   0   0   0   0   0   0  ...   0   0   0   0   0   0   0   
698   0   0   0   0   0   0   0   0   0   0  ...   0   0   0   0   1   0   0   
699   0   0   0   0   0   0   0   0   0   0  ...   0   0   0   0   0   0   0   

     26  27  28  
0     1   0   0  
1     0   0   0  
2     0   0   0  
3     0   0   0  
4     0   0   0  
..   ..  ..  ..  
695   0   0   0  
696   0   0   0  
697   0   0   0  
698   0   0   0  
699   0   0   1  

[700 rows x 28 columns]

In [14]:
# To check whether have rows with missing values, if check = 0(have missing value), 
# if check = 1(only a single value in the row)
reason_columns['check'] = reason_columns.sum(axis=1)

In [15]:
reason_columns

0  1  2  3  4  5  6  7  8  9  ...  19  21  22  23  24  25  26  27  28  \
0    0  0  0  0  0  0  0  0  0  0  ...   0   0   0   0   0   0   1   0   0   
1    1  0  0  0  0  0  0  0  0  0  ...   0   0   0   0   0   0   0   0   0   
2    0  0  0  0  0  0  0  0  0  0  ...   0   0   0   1   0   0   0   0   0   
3    0  0  0  0  0  0  0  1  0  0  ...   0   0   0   0   0   0   0   0   0   
4    0  0  0  0  0  0  0  0  0  0  ...   0   0   0   1   0   0   0   0   0   
..  .. .. .. .. .. .. .. .. .. ..  ...  ..  ..  ..  ..  ..  ..  ..  ..  ..   
695  0  0  0  0  0  0  0  0  0  0  ...   0   0   0   0   0   0   0   0   0   
696  0  0  0  0  0  0  1  0  0  0  ...   0   0   0   0   0   0   0   0   0   
697  0  0  0  0  0  0  0  0  0  0  ...   0   0   0   0   0   0   0   0   0   
698  0  0  0  0  0  0  0  0  0  0  ...   0   0   0   1   0   0   0   0   0   
699  0  0  0  0  0  0  0  0  0  0  ...   0   0   0   0   0   0   0   0   1   

     check  
0        1  
1        1  
2        1  
3        1  
4        1  
..     ...  
695      1  
696      1  
697      1  
698      1  
699      1  

[700 rows x 29 columns]

In [16]:
# To check how many different values in the 'check' column
reason_columns['check'].unique()

array([1])

Means all values in the 'check' column are '1' (no missing values or incorrect values)

In [17]:
# To remove the 'check' column after validation
reason_columns = reason_columns.drop(['check'], axis=1)
reason_columns

0   1   2   3   4   5   6   7   8   9   ...  18  19  21  22  23  24  25  \
0     0   0   0   0   0   0   0   0   0   0  ...   0   0   0   0   0   0   0   
1     1   0   0   0   0   0   0   0   0   0  ...   0   0   0   0   0   0   0   
2     0   0   0   0   0   0   0   0   0   0  ...   0   0   0   0   1   0   0   
3     0   0   0   0   0   0   0   1   0   0  ...   0   0   0   0   0   0   0   
4     0   0   0   0   0   0   0   0   0   0  ...   0   0   0   0   1   0   0   
..   ..  ..  ..  ..  ..  ..  ..  ..  ..  ..  ...  ..  ..  ..  ..  ..  ..  ..   
695   0   0   0   0   0   0   0   0   0   0  ...   0   0   0   0   0   0   0   
696   0   0   0   0   0   0   1   0   0   0  ...   0   0   0   0   0   0   0   
697   0   0   0   0   0   0   0   0   0   0  ...   0   0   0   0   0   0   0   
698   0   0   0   0   0   0   0   0   0   0  ...   0   0   0   0   1   0   0   
699   0   0   0   0   0   0   0   0   0   0  ...   0   0   0   0   0   0   0   

     26  27  28  
0     1   0   0  
1     0   0   0  
2     0   0   0  
3     0   0   0  
4     0   0   0  
..   ..  ..  ..  
695   0   0   0  
696   0   0   0  
697   0   0   0  
698   0   0   0  
699   0   0   1  

[700 rows x 28 columns]

In [18]:
# To drop the Reason 0 dummy variable column to avoid potential multicollinearity issues 
reason_columns = pd.get_dummies(df['Reason for Absence'], drop_first = True)

In [19]:
reason_columns

1   2   3   4   5   6   7   8   9   10  ...  18  19  21  22  23  24  25  \
0     0   0   0   0   0   0   0   0   0   0  ...   0   0   0   0   0   0   0   
1     0   0   0   0   0   0   0   0   0   0  ...   0   0   0   0   0   0   0   
2     0   0   0   0   0   0   0   0   0   0  ...   0   0   0   0   1   0   0   
3     0   0   0   0   0   0   1   0   0   0  ...   0   0   0   0   0   0   0   
4     0   0   0   0   0   0   0   0   0   0  ...   0   0   0   0   1   0   0   
..   ..  ..  ..  ..  ..  ..  ..  ..  ..  ..  ...  ..  ..  ..  ..  ..  ..  ..   
695   0   0   0   0   0   0   0   0   0   1  ...   0   0   0   0   0   0   0   
696   0   0   0   0   0   1   0   0   0   0  ...   0   0   0   0   0   0   0   
697   0   0   0   0   0   0   0   0   0   1  ...   0   0   0   0   0   0   0   
698   0   0   0   0   0   0   0   0   0   0  ...   0   0   0   0   1   0   0   
699   0   0   0   0   0   0   0   0   0   0  ...   0   0   0   0   0   0   0   

     26  27  28  
0     1   0   0  
1     0   0   0  
2     0   0   0  
3     0   0   0  
4     0   0   0  
..   ..  ..  ..  
695   0   0   0  
696   0   0   0  
697   0   0   0  
698   0   0   0  
699   0   0   1  

[700 rows x 27 columns]

# Group the Reasons for Absence:

In [20]:
# To get a list of column names
df.columns.values

array(['Reason for Absence', 'Date', 'Transportation Expense',
       'Distance to Work', 'Age', 'Daily Work Load Average',
       'Body Mass Index', 'Education', 'Children', 'Pets',
       'Absenteeism Time in Hours'], dtype=object)

In [21]:
df = df.drop(['Reason for Absence'], axis = 1)

In [22]:
df

Date  Transportation Expense  Distance to Work  Age  \
0    07/07/2015                     289                36   33   
1    14/07/2015                     118                13   50   
2    15/07/2015                     179                51   38   
3    16/07/2015                     279                 5   39   
4    23/07/2015                     289                36   33   
..          ...                     ...               ...  ...   
695  23/05/2018                     179                22   40   
696  23/05/2018                     225                26   28   
697  24/05/2018                     330                16   28   
698  24/05/2018                     235                16   32   
699  31/05/2018                     291                31   40   

     Daily Work Load Average  Body Mass Index  Education  Children  Pets  \
0                    239.554               30          1         2     1   
1                    239.554               31          1         1     0   
2                    239.554               31          1         0     0   
3                    239.554               24          1         2     0   
4                    239.554               30          1         2     1   
..                       ...              ...        ...       ...   ...   
695                  237.656               22          2         2     0   
696                  237.656               24          1         1     2   
697                  237.656               25          2         0     0   
698                  237.656               25          3         0     0   
699                  237.656               25          1         1     1   

     Absenteeism Time in Hours  
0                            4  
1                            0  
2                            2  
3                            4  
4                            2  
..                         ...  
695                          8  
696                          3  
697                          8  
698                          2  
699                          2  

[700 rows x 10 columns]

In [23]:
reason_columns.loc[:, 1:14].max(axis=1)

0      0
1      0
2      0
3      1
4      0
      ..
695    1
696    1
697    1
698    0
699    0
Length: 700, dtype: uint8

# Based on the Absenteeism Feature Descriptions Table
We can extract the meaning from those numeric values and group those into different groups:

    - Reason 0(no reason): the baseline model(when no reason is given)
    - Reason_1: various diseases
    - Reason_2: pregnancy and giving birth
    - Reason_3: poisoning and peculiar reasons not categorized elsewhere 
    - Reason_4: light diseases

In [24]:
# Create 4 new columns for the reason of absence
reason_type_1 = reason_columns.loc[:, 1:14].max(axis=1)
reason_type_2 = reason_columns.loc[:, 15:17].max(axis=1)
reason_type_3 = reason_columns.loc[:, 18:21].max(axis=1)
reason_type_4 = reason_columns.loc[:, 22:].max(axis=1)

# Concatenate Column Values

In [25]:
# To add 4 new created columns to the dataframe
df = pd.concat([df, reason_type_1, reason_type_2, reason_type_3, reason_type_4], axis = 1)

In [26]:
df

Date  Transportation Expense  Distance to Work  Age  \
0    07/07/2015                     289                36   33   
1    14/07/2015                     118                13   50   
2    15/07/2015                     179                51   38   
3    16/07/2015                     279                 5   39   
4    23/07/2015                     289                36   33   
..          ...                     ...               ...  ...   
695  23/05/2018                     179                22   40   
696  23/05/2018                     225                26   28   
697  24/05/2018                     330                16   28   
698  24/05/2018                     235                16   32   
699  31/05/2018                     291                31   40   

     Daily Work Load Average  Body Mass Index  Education  Children  Pets  \
0                    239.554               30          1         2     1   
1                    239.554               31          1         1     0   
2                    239.554               31          1         0     0   
3                    239.554               24          1         2     0   
4                    239.554               30          1         2     1   
..                       ...              ...        ...       ...   ...   
695                  237.656               22          2         2     0   
696                  237.656               24          1         1     2   
697                  237.656               25          2         0     0   
698                  237.656               25          3         0     0   
699                  237.656               25          1         1     1   

     Absenteeism Time in Hours  0  1  2  3  
0                            4  0  0  0  1  
1                            0  0  0  0  0  
2                            2  0  0  0  1  
3                            4  1  0  0  0  
4                            2  0  0  0  1  
..                         ... .. .. .. ..  
695                          8  1  0  0  0  
696                          3  1  0  0  0  
697                          8  1  0  0  0  
698                          2  0  0  0  1  
699                          2  0  0  0  1  

[700 rows x 14 columns]

In [27]:
df.columns.values

array(['Date', 'Transportation Expense', 'Distance to Work', 'Age',
       'Daily Work Load Average', 'Body Mass Index', 'Education',
       'Children', 'Pets', 'Absenteeism Time in Hours', 0, 1, 2, 3],
      dtype=object)

In [28]:
# To assign the column names to the dataframe
column_names = ['Date', 'Transportation Expense', 'Distance to Work', 'Age',
       'Daily Work Load Average', 'Body Mass Index', 'Education',
       'Children', 'Pets', 'Absenteeism Time in Hours', 'Reason_1', 'Reason_2', 'Reason_3', 'Reason_4']

In [29]:
df.columns = column_names

In [181]:
df.head()

Date  Transportation Expense  Distance to Work  Age  \
0  07/07/2015                     289                36   33   
1  14/07/2015                     118                13   50   
2  15/07/2015                     179                51   38   
3  16/07/2015                     279                 5   39   
4  23/07/2015                     289                36   33   

   Daily Work Load Average  Body Mass Index  Education  Children  Pets  \
0                  239.554               30          1         2     1   
1                  239.554               31          1         1     0   
2                  239.554               31          1         0     0   
3                  239.554               24          1         2     0   
4                  239.554               30          1         2     1   

   Absenteeism Time in Hours  Reason_1  Reason_2  Reason_3  Reason_4  
0                          4         0         0         0         1  
1                          0         0         0         0         0  
2                          2         0         0         0         1  
3                          4         1         0         0         0  
4                          2         0         0         0         1

# Reorder Columns

In [30]:
# To move the Reasons at the beginning
column_names_reordered =['Reason_1', 'Reason_2', 'Reason_3', 'Reason_4', 
                         'Date', 'Transportation Expense', 'Distance to Work', 'Age',
       'Daily Work Load Average', 'Body Mass Index', 'Education',
       'Children', 'Pets', 'Absenteeism Time in Hours']

In [31]:
df = df[column_names_reordered]

In [32]:
df.head()

Reason_1  Reason_2  Reason_3  Reason_4        Date  Transportation Expense  \
0         0         0         0         1  07/07/2015                     289   
1         0         0         0         0  14/07/2015                     118   
2         0         0         0         1  15/07/2015                     179   
3         1         0         0         0  16/07/2015                     279   
4         0         0         0         1  23/07/2015                     289   

   Distance to Work  Age  Daily Work Load Average  Body Mass Index  Education  \
0                36   33                  239.554               30          1   
1                13   50                  239.554               31          1   
2                51   38                  239.554               31          1   
3                 5   39                  239.554               24          1   
4                36   33                  239.554               30          1   

   Children  Pets  Absenteeism Time in Hours  
0         2     1                          4  
1         1     0                          0  
2         0     0                          2  
3         2     0                          4  
4         2     1                          2

# Date

In [33]:
# To check the type of 'Date'
type(df['Date'][0])

str

In [34]:
# To convert the 'Date' type from 'str' to 'timestamp'
df['Date'] = pd.to_datetime(df['Date'], format = '%d/%m/%Y')

In [35]:
df['Date']

0     2015-07-07
1     2015-07-14
2     2015-07-15
3     2015-07-16
4     2015-07-23
         ...    
695   2018-05-23
696   2018-05-23
697   2018-05-24
698   2018-05-24
699   2018-05-31
Name: Date, Length: 700, dtype: datetime64[ns]

In [36]:
# To extract the Month Value
df['Date'][0].month

7

In [37]:
# Assign an empty list
list_months = []
list_months

[]

In [38]:
# To attach the new value obtained from each iteration to the list of 'list_month'
for i in range(700):
    list_months.append(df['Date'][i].month)

In [39]:
list_months

[7,
 7,
 7,
 7,
 7,
 7,
 7,
 7,
 7,
 7,
 7,
 7,
 7,
 7,
 7,
 7,
 7,
 7,
 7,
 8,
 8,
 8,
 8,
 8,
 8,
 8,
 8,
 8,
 8,
 8,
 8,
 8,
 8,
 8,
 8,
 8,
 8,
 8,
 8,
 8,
 9,
 9,
 9,
 9,
 9,
 9,
 9,
 9,
 9,
 9,
 9,
 9,
 9,
 9,
 9,
 9,
 9,
 9,
 9,
 9,
 9,
 9,
 10,
 10,
 10,
 10,
 10,
 10,
 10,
 10,
 10,
 10,
 10,
 10,
 10,
 10,
 10,
 10,
 10,
 10,
 11,
 11,
 11,
 11,
 11,
 11,
 11,
 11,
 11,
 11,
 11,
 11,
 11,
 11,
 11,
 11,
 11,
 11,
 12,
 12,
 12,
 12,
 12,
 12,
 12,
 12,
 12,
 12,
 12,
 12,
 12,
 12,
 12,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 2,
 2,
 2,
 2,
 2,
 2,
 2,
 2,
 2,
 2,
 2,
 2,
 2,
 2,
 2,
 2,
 2,
 2,
 3,
 3,
 3,
 3,
 3,
 3,
 3,
 3,
 3,
 3,
 3,
 3,
 3,
 3,
 3,
 3,
 3,
 3,
 3,
 3,
 3,
 3,
 3,
 3,
 3,
 3,
 3,
 3,
 3,
 4,
 4,
 4,
 4,
 4,
 4,
 4,
 4,
 4,
 4,
 4,
 4,
 4,
 4,
 4,
 4,
 4,
 4,
 4,
 4,
 5,
 5,
 5,
 5,
 5,
 5,
 5,
 5,
 5,
 5,
 5,
 5,
 5,
 5,
 5,
 5,
 6,
 6,
 6,
 6,
 6,
 6,
 6,
 6,
 6,
 6,
 6,
 6,
 6,
 6,
 6,
 6,
 7,
 

In [40]:
# To create a new column 'Month Value'
df['Month Value'] = list_months

In [41]:
df.head(20)

Reason_1  Reason_2  Reason_3  Reason_4       Date  Transportation Expense  \
0          0         0         0         1 2015-07-07                     289   
1          0         0         0         0 2015-07-14                     118   
2          0         0         0         1 2015-07-15                     179   
3          1         0         0         0 2015-07-16                     279   
4          0         0         0         1 2015-07-23                     289   
5          0         0         0         1 2015-07-10                     179   
6          0         0         0         1 2015-07-17                     361   
7          0         0         0         1 2015-07-24                     260   
8          0         0         1         0 2015-07-06                     155   
9          0         0         0         1 2015-07-13                     235   
10         1         0         0         0 2015-07-20                     260   
11         1         0         0         0 2015-07-14                     260   
12         1         0         0         0 2015-07-15                     260   
13         1         0         0         0 2015-07-15                     179   
14         0         0         0         1 2015-07-15                     179   
15         1         0         0         0 2015-07-17                     246   
16         0         0         0         1 2015-07-17                     179   
17         0         0         1         0 2015-07-27                     179   
18         1         0         0         0 2015-07-30                     189   
19         0         0         0         1 2015-08-05                     248   

    Distance to Work  Age  Daily Work Load Average  Body Mass Index  \
0                 36   33                  239.554               30   
1                 13   50                  239.554               31   
2                 51   38                  239.554               31   
3                  5   39                  239.554               24   
4                 36   33                  239.554               30   
5                 51   38                  239.554               31   
6                 52   28                  239.554               27   
7                 50   36                  239.554               23   
8                 12   34                  239.554               25   
9                 11   37                  239.554               29   
10                50   36                  239.554               23   
11                50   36                  239.554               23   
12                50   36                  239.554               23   
13                51   38                  239.554               31   
14                51   38                  239.554               31   
15                25   41                  239.554               23   
16                51   38                  239.554               31   
17                51   38                  239.554               31   
18                29   33                  239.554               25   
19                25   47                  205.917               32   

    Education  Children  Pets  Absenteeism Time in Hours  Month Value  
0           1         2     1                          4            7  
1           1         1     0                          0            7  
2           1         0     0                          2            7  
3           1         2     0                          4            7  
4           1         2     1                          2            7  
5           1         0     0                          2            7  
6           1         1     4                          8            7  
7           1         4     0                          4            7  
8           1         2     0                         40            7  
9           3         1     1                          8            7  
10          1     

In [42]:
df['Date'][699].weekday()

3

In [43]:
# To define a function to return the weekday from date column
def date_to_weekday(date_value):
    return date_value.weekday()

In [44]:
# To create a 'Day of the Week' column 
df['Day of the Week'] = df['Date'].apply(date_to_weekday)

In [45]:
df.head()

Reason_1  Reason_2  Reason_3  Reason_4       Date  Transportation Expense  \
0         0         0         0         1 2015-07-07                     289   
1         0         0         0         0 2015-07-14                     118   
2         0         0         0         1 2015-07-15                     179   
3         1         0         0         0 2015-07-16                     279   
4         0         0         0         1 2015-07-23                     289   

   Distance to Work  Age  Daily Work Load Average  Body Mass Index  Education  \
0                36   33                  239.554               30          1   
1                13   50                  239.554               31          1   
2                51   38                  239.554               31          1   
3                 5   39                  239.554               24          1   
4                36   33                  239.554               30          1   

   Children  Pets  Absenteeism Time in Hours  Month Value  Day of the Week  
0         2     1                          4            7                1  
1         1     0                          0            7                1  
2         0     0                          2            7                2  
3         2     0                          4            7                3  
4         2     1                          2            7                3

In [46]:
# To check the 'Transportation Expense' data type
type(df['Transportation Expense'][0])

numpy.int64

In [47]:
# To check the 'Distance to Work' data type
type(df['Distance to Work'][0])

numpy.int64

In [49]:
# To check the 'Age' data type
type(df['Age'][0])

numpy.int64

In [50]:
# To check the 'Daily Work Load Average' data type
type(df['Daily Work Load Average'][0])

numpy.float64

In [51]:
# To check the 'Body Mass Index' data type
type(df['Body Mass Index'][0])

numpy.int64

# Education

In [53]:
# To check how many unique values in 'Education' column
df['Education'].unique()

array([1, 3, 2, 4])

###### Interpretation for the Values in 'Education' column:
    1: High School
    2: Graduate
    3: Postgraduate
    4: A master or a Doctor

In [54]:
# To check each value in 'Education' column appeared how many times
df['Education'].value_counts()

1    583
3     73
2     40
4      4
Name: Education, dtype: int64

##### As 583 people have high school education only, and 117 people have a better degree than that. Therefore, we can combine value 3, 2, 4 in a single category.

In [55]:
# To assign new values to the values in 'Education' column, and combine value 3, 2, 4 in a single category.
df['Education'] = df['Education'].map({1:0, 3:1, 2:1, 4:1})

In [56]:
# To check the how many unique values in 'Education' column after assign new values
df['Education'].unique()

array([0, 1])

In [57]:
# To check how many people have high school education only, and how many people have a better degree than that.
df['Education'].value_counts()

0    583
1    117
Name: Education, dtype: int64

# Create a Checkpoint

In [59]:
# To reduce the risk of losing important data
df_preprocessed = df.copy()

In [60]:
df_preprocessed.head(10)

Reason_1  Reason_2  Reason_3  Reason_4       Date  Transportation Expense  \
0         0         0         0         1 2015-07-07                     289   
1         0         0         0         0 2015-07-14                     118   
2         0         0         0         1 2015-07-15                     179   
3         1         0         0         0 2015-07-16                     279   
4         0         0         0         1 2015-07-23                     289   
5         0         0         0         1 2015-07-10                     179   
6         0         0         0         1 2015-07-17                     361   
7         0         0         0         1 2015-07-24                     260   
8         0         0         1         0 2015-07-06                     155   
9         0         0         0         1 2015-07-13                     235   

   Distance to Work  Age  Daily Work Load Average  Body Mass Index  Education  \
0                36   33                  239.554               30          0   
1                13   50                  239.554               31          0   
2                51   38                  239.554               31          0   
3                 5   39                  239.554               24          0   
4                36   33                  239.554               30          0   
5                51   38                  239.554               31          0   
6                52   28                  239.554               27          0   
7                50   36                  239.554               23          0   
8                12   34                  239.554               25          0   
9                11   37                  239.554               29          1   

   Children  Pets  Absenteeism Time in Hours  Month Value  Day of the Week  
0         2     1                          4            7                1  
1         1     0                          0            7                1  
2         0     0                          2            7                2  
3         2     0                          4            7                3  
4         2     1                          2            7                3  
5         0     0                          2            7                4  
6         1     4                          8            7                4  
7         4     0                          4            7                4  
8         2     0                         40            7                0  
9         1     1                          8            7                0

In [61]:
df_preprocessed.to_csv('Absenteeism_preprocessed.csv', index=False)

# Notes

Why drop ID column: ID is the individual identification of each person, and each number from ID column represents a specific employee, indicates precisely who has been away during working hours, ID information could help company management when monitoring the presence of each employee. However, ID information will not improve the analysis as it is an nominal data and it not carrys any numeric information and it will harm the precision of estimations


quantitative analysis: add numeric meaning to categorical nominal values
   * Way to do it: Turn the nominal values into dummy variables ( dummy variable: an explanatory binary variable that equals 1 if a certain categorical effect is present, and that equals 0 if that same effect is absent)
   * Therefore, we want a column where the values of 1 will appear in case an individual has been absent because of reason number 1, and 0 if she was absent because of another reason